tiledb/sm/query/query.cc

```diff
@@ -1885,28 +1885,35 @@ Status Query::set_subarray_unsafe(const NDRange& subarray) {
 Status Query::check_buffers_correctness() {
   // Iterate through each attribute
   for (auto& attr : buffer_names()) {
-    if (buffer(attr).buffer_ == nullptr)
-      return logger_->status(Status::QueryError(
-          std::string("Data buffer is not set for " + attr)));
     if (array_schema_->var_size(attr)) {
       // Var-sized
-      // Check for data buffer under buffer_var
-      bool exists_data = buffer(attr).buffer_var_ != nullptr;
-      bool exists_offset = buffer(attr).buffer_ != nullptr;
-      if ((!exists_data && *buffer(attr).buffer_var_size_ != 0) ||
-          !exists_offset) {
+      // Check for data buffer under buffer_var and offsets buffer under buffer
+      if (buffer(attr).buffer_var_ == nullptr &&
+          *buffer(attr).buffer_var_size_ != 0) {
+        return logger_->status(Status::QueryError(
+            std::string("Var-Sized input attribute/dimension '") + attr +
+            "' is not set correctly. \nData buffer is not set."));
+      }
+      if (buffer(attr).buffer_ == nullptr) {
         return logger_->status(Status::QueryError(
             std::string("Var-Sized input attribute/dimension '") + attr +
-            "' is not set correctly \nOffsets buffer is not set"));
+            "' is not set correctly. \nOffsets buffer is not set."));
       }
+      if (buffer(attr).buffer_ != nullptr &&
```

Comment:
The first check is redundant, whereas the second and third are a repetition from above. What do you want to do here?